controls/cGameControlsContext.h

```diff
@@ -17,13 +10,48 @@
 //
 // a context belongs to a player
 
-class cGameControlsContext {
+
+// 24/12/2021 -> This is a kind of "Combat Mouse State" object really. This could move into a
+// combat state object, or 2 separate objects (state & mouse object).
+
+// Mouse status (taken from somewhere else)
+//#define MOUSE_STATE_NORMAL  0
+//#define MOUSE_STATE_MOVE    1
+//#define MOUSE_STATE_ATTACK  2
+//#define MOUSE_STATE_PLACE   3
+
+#include "controls/mousestates/cMouseNormalState.h"
+#include "controls/mousestates/cMouseUnitsSelectedState.h"
+#include "controls/mousestates/cMouseRepairState.h"
+
+enum eMouseState {
+    MOUSESTATE_SELECT, // mouse is in 'normal' mode, it is able to select units
+    MOUSESTATE_UNITS_SELECTED, // mouse has selected units
+    MOUSESTATE_REPAIR, // mouse is in repair state
+    MOUSESTATE_PLACE, // mouse is going to place a structure
+};
+
+static const char* mouseStateString(const eMouseState &state) {

```

Comment:
```suggestion
static std::string mouseStateString(eMouseState state) {
```
Returning a string is better practice than a raw character pointer. Then the caller doesn't need to worry about the pointer being nullptr or pointing to something that's cleaned up or overwritten. And in modern compilers, we can rely on RVO (return value optimization) so that the string is created in-place and there is no copy (or move).

The eMouseState enum is so small, the advice is to pass it by copy. It's smaller than a pointer, so it can be passed more efficiently (in a CPU register, for example). An extensive coverage of that topic is here: https://www.modernescpp.com/index.php/c-core-guidelines-how-to-pass-function-parameters

Reply:
Ah this is good stuff. I kind of try to make it my best practice to have `const <type> & var` in method calls. So I guess this depends on the size of `type` then. 

About `std::string` thx! :)